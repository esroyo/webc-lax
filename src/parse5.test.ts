import { parse5 } from './index.ts';
import { assertSnapshot } from '@std/testing/snapshot';

Deno.test('should keep slot attrs, contents and position in the HTML (head or body)', async (t) => {
    const parsed = parse5.parse(`<html>
  <head>
    <title>My website</title>
    <slot name="my-styles"><style>* { color: inherit; }</style></slot>
  </head>
  <body>
    <slot>My default content</slot>
    <slot name="my-js"></slot>
  </body>
</html>
`);
    const html = parse5.serialize(parsed);
    await assertSnapshot(t, html);
});
