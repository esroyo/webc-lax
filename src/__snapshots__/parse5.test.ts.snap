export const snapshot = {};

snapshot[`should keep slot attrs, contents and position in the HTML (head or body) 1`] = `
'<html><head>
    <custom-title>My website</custom-title>
    <slot name="my-styles"><style>* { color: inherit; }</style></slot>
  </head>
  <body>
    <slot>My default content</slot>
    <slot name="my-js"></slot>
  

</body></html>'
`;
